.github/PULL_REQUEST_TEMPLATE

```diff
@@ -1,10 +1,32 @@
-## What changes were proposed in this pull request?
+<!--  Thanks for sending a pull request!  Here are some tips for you:
+1. If this is your first time, please read our contributor guidelines: https://spark.apache.org/contributing.html
+2. Ensure you have added or run the appropriate tests for your PR: https://spark.apache.org/developer-tools.html
+3. Understand the relationship beteween JIRAs to file issues and PRs to propose some changes for them, as documented in guidelines.
+4. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][SPARK-XXXX] Your PR title ...'
+5. Update your PR description whenever it is needed.
+6. Provide a minimised example to reproduce the issue if possible.
+-->
 
-(Please fill in changes proposed in this fix)
+### What changes were proposed in this pull request?
 
-## How was this patch tested?
+### Why changes are needed?
 
-(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
-(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)
+### Special notes for your reviewer
+<!--
+This special notes can be any notes for reviewers to review faster and more efficiently. For instance, see the examples below.
+1. If you refactor some codes with changing classes, showing the class hierarchy will help reviewers.
+2. If you fix some SQL features, you can provide some references of other DBMSes.
+3. If there is a design documentation, please leave the link.
+4. If there is a discussion in the mailing thread, please leave the link.
+-->
 
-Please review https://spark.apache.org/contributing.html before opening a pull request.
+### Does this PR introduce any user-facing change?
+<!--
+If yes, please clarify the previous behavior and the change this PR proposes. Also, please provide the console output, description and/or the reproducer to test if possible so that reviewers can review this patch faster.
+If no, write 'No'.
+-->
+
+### How was this patch tested?
+<!--
+If it was tested in a way different from regular unit tests, please clarify how you tested step by step, ideally copy and paste-able, so that other reviewers can test and check through, and descendants can verify in the future.
```

Comment:
If only a few test suites check my changes, should I mention those suites here, or I can skip the entire section?

Reply:
I think it's just like we did before - let me clarify it.

One thing I would like to emphasize here was to leave a reproducer to show how to test step by step if tests are not added or not enough. Some people still say just "Manually tested" here often.

Reply:
In most cases, unit tests are not enough. Most PRs should also include end-to-end tests and perf tests. Also, I found most of new PR authors just write the positive cases. We should remind them to add negative test cases too.

Reply:
That's right in general, in particular, for SQL test cases but I think it might be different in other components for many reasons. For instance, difficult to write a negative test case, etc. Let me try to clarify it in a general way.

I think we should write things that all people here could generally agree upon for now.